### PR TITLE
Fix functions affected by manual donor

### PR DESCRIPTION
**Pull request by yedidya03**

### CODE REVIEW

File: packages/functions/src/coordinator/GetDonorsHandler.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ export default async function (
   request: FunctionsApi.GetDonorsRequest,
   callerId: string
 ): Promise<FunctionsApi.GetDonorsResponse> {
+  // TODO : should it return also manual donor that this coordinator regisered?
```
Comment (yaron1m) [inline packages/functions/src/coordinator/GetDonorsHandler.ts:9]: I think not. They are not real entities

File: packages/functions/src/jobs/CronJobs.ts [hunk 2]
```diff
@@ -52,12 +56,16 @@ export const SendConfirmationReminders = async (
...
   const donorIds = appointments.map((appointment) => appointment.donorId);
 
-  const donorsInAppointments = await getDonors(_.uniq(donorIds));
+  const donorsInAppointments: { [donorId: string]: DbDonor } = (
+    await getDonors(_.uniq(donorIds))
+  ).reduce((obj, donor: DbDonor) => {
+    return { ...obj, [donor.id]: donor };
+  }, {});
+
+  delete donorsInAppointments[MANUAL_DONOR_ID];
```
Comment (yaron1m) [inline packages/functions/src/jobs/CronJobs.ts:65]: Use `donorIds.filter(id => id  !=== MANUAL_DONOR_ID)`

Comment (yaron1m) [inline packages/functions/src/jobs/CronJobs.ts:65]: Use `donorIds.filter(id => id  !=== MANUAL_DONOR_ID)`

Comment (yaron1m) [inline packages/functions/src/jobs/CronJobs.ts:65]: Use `donorIds.filter(id => id  !=== MANUAL_DONOR_ID)`

Comment (yedidya03) [reply packages/functions/src/jobs/CronJobs.ts:65]: I added this code for improving time complexity in line 68..

Comment (yedidya03) [reply packages/functions/src/jobs/CronJobs.ts:65]: however maybe it worth adding a comment of what those lines do..

File: packages/functions/src/coordinator/GetDonorsHandler.ts (hunk 1 above)
Comment (yedidya03) [reply packages/functions/src/coordinator/GetDonorsHandler.ts:9]: 👍

File: packages/functions/src/jobs/CronJobs.ts (hunk 2 above)
Comment (yaron1m) [reply packages/functions/src/jobs/CronJobs.ts:65]: It is great, just don't use `delete`. 
Also, there's no need for `reduce`.

`.filter` will get you what you want, no?

Comment (yedidya03) [reply packages/functions/src/jobs/CronJobs.ts:65]: In line 68, I want to get the DBDonor object from the donorId that I get from the appointment.
fot that, I need a dictionary like object that the key is the donorId and the value is the donor object, in that way I can get the donor object from the donorId in O(1) complexity.
so what I need to do is to build this dictionary..
what I have for building this dictionary is an array of DBDonor objects (from getDonor(...)) and inside eatch donor object there is a donorId that should turn to the key of the item in the dictionary.
filter will return an array, and that is not what I need. with reduce I can build this dictionary in an easy way.

Comment (yedidya03) [reply packages/functions/src/jobs/CronJobs.ts]: about the delete, you are probebly write becouse the getDonors() function will not return items with id of MANUAL_DONOR_ID..

Comment (yaron1m) [reply packages/functions/src/jobs/CronJobs.ts]: So why not use `map`?

```
const donorsInAppointments = await getDonors(_.uniq(donorIds));
const donorsMap : { [donorId: string]: DbDonor};
donorsInAppointments.map(donor => donorsMap[donor.id]=donor } 
```

**Review by yaron1m — APPROVED** [review]

**Pull request merged**